lightningd/lightningd.c

```diff
@@ -757,13 +757,15 @@ static int setup_sig_handlers(void)
 
 /*~ This removes the SIGCHLD handler, so we don't try to write
  * to a broken pipe. */
-static void remove_sigchild_handler(void)
+static void remove_sigchild_handler(int sigchld_rfd)
 {
 	struct sigaction sigchild;
 
 	memset(&sigchild, 0, sizeof(struct sigaction));
 	sigchild.sa_handler = SIG_DFL;
 	sigaction(SIGCHLD, &sigchild, NULL);
+	close(sigchld_wfd);
+	close(sigchld_rfd);
```

Comment:
Better to io_close() the io_conn associated with sigchld_rfd rather than just mug the fd.

i.e. save the pointer here:

```
notleak(io_new_conn(ld, sigchld_rfd, sigchld_rfd_in, ld));
```

And pass *that* into remove_sigchild_handler() to close.